src/client.c

```diff
@@ -97,12 +144,53 @@ int run_online(char *host, int port, char *username, char *opponent) {
 	return EXIT_SUCCESS;
 }
 
+/**
+ * Show the main menu for gameplay selection
+ */
+int main_menu(char *host, int port) {
+	WidgetSelection *selection;
+	int selected_choice;
+	char *choices[] = {
+	    "Single", "Multi", "Controls", "Exit", (char *)NULL,
+	};
+
+	while (1) {
+		// TODO We shouldn't need to be calling all this setup
+		// everytime. The main change is that the init_pair calls are
+		// different for the gameplay vs. the menu
+		initscr();
+		start_color();
+		cbreak();
+		noecho();
+		keypad(stdscr, TRUE);
+		init_pair(1, COLOR_RED, COLOR_BLACK);
+		selection = ttviz_select(choices, 4, "Gameplay Mode", 1);
+		selected_choice = selection_to_index(selection);
+
+		switch (selected_choice) {
+		case 0:
+			run_offline();
+			break;
+		case 1:
+			run_online(host, port);
+			break;
+		case 2:
```

Comment:
Add a TODO here, And we should open up an issue for this.

Reply:
Good point. Implementing this can go under #21.